internal/clientconn/conninfo/conn_info.go

```diff
@@ -53,6 +65,27 @@ func (connInfo *ConnInfo) SetAuth(username, password string) {
 	connInfo.password = password
 }
 
+// Cursor returns the cursor value stored.
+func (connInfo *ConnInfo) Cursor(collection string) iterator.Interface[int, any] {
+	return connInfo.cursor[collection]
+}
+
+// SetCursor stores the cursor value.
+func (connInfo *ConnInfo) SetCursor(collection string, cursor iterator.Interface[int, any]) {
+	connInfo.curRW.Lock()
+	defer connInfo.curRW.Unlock()
+
+	connInfo.cursor[collection] = cursor
+}
+
+// RemoveCursor removes the cursor value stored.
+func (connInfo *ConnInfo) RemoveCursor(collection string) {
```

Comment:
Where do we call it from?

Reply:
We call it from `MsgGetMore` when the iterator is done.